lib/oauth2/client.rb

```diff
@@ -62,7 +63,8 @@ def connection
     # The authorize endpoint URL of the OAuth2 provider
     #
     # @param [Hash] params additional query parameters
-    def authorize_url(params = nil)
+    def authorize_url(params = {})
+      params = (params || {}).merge(redirection_params)
```

Comment:
I wonder if it is more correct (spec-wise) for these redirection params to only be added to `AuthCode#authorize_params` instead.